docs/intro/using.rst

```diff
@@ -86,47 +85,94 @@ is set using the utility function :meth:`~hybrid.utils.min_sample`.
 ...           SplatComposer())
 >>> new_state = branch.next(State.from_sample(min_sample(bqm), bqm))
 >>> print(new_state.subsamples)      # doctest: +SKIP
-       4   5   6   7   8   9  energy  num_occ.
-   0  +1  -1  -1  +1  -1  +1    -5.0         1
-   1  +1  -1  -1  +1  -1  +1    -5.0         1
-
-   [ 2 rows, 6 variables ]
+    4   5   6   7   8   9  energy  num_occ.
+0  +1  -1  -1  +1  -1  +1    -5.0         1
+1  +1  -1  -1  +1  -1  +1    -5.0         1
+['SPIN', 2 rows, 2 samples, 6 variables]
 
 Such :class:`.Branch` classes can be run in parallel using the :class:`.RacingBranches` class.
 From the outputs of these parallel branches, :class:`.ArgMin` selects a new current sample.
 And instead of a single iteration on the sample set, you can use the :class:`.Loop`
 to iterate a set number of times or until a convergence criteria is met.
 
-This next example solves a binary quadratic model by iteratively producing best samples.
+This example of :ref:`racingBranches1` solves a binary quadratic model by iteratively producing best samples.
 Similar to :std:doc:`qbsolv <qbsolv:index>`, it employs both tabu search on the entire
 problem and a D-Wave system on subproblems. In addition to building-block components
 such as employed above, this example also uses infrastructure classes to manage the
 decomposition and parallel running of branches.
 
+.. figure:: ../_static/racing_branches_1.png
+  :name: racingBranches1
+  :scale: 90 %
+  :alt: Racing Branches
+
+  Racing Branches
 
 .. include:: ../../README.rst
   :start-after: example-start-marker
   :end-before: example-end-marker
 
+Flow Refining
+-------------
+
+The framework enables quick modification of work flows to improve solutions and performance.
+For example, the :ref:`racingBranches1` workflow above is modified by the code below to
+better fit problems with large numbers of variables: setting `rolling_history` moves a
+decomposition window along a fraction of problem variables as ordered by energy impact.
+
+.. code-block:: python
+
+    # Redefine the workflow: a rolling decomposition window
+    subproblem = hybrid.EnergyImpactDecomposer(size=50, rolling_history=0.15)
+    subsampler = hybrid.QPUSubproblemAutoEmbeddingSampler()
+                 | hybrid.SplatComposer()
+
+    iteration = hybrid.RacingBranches(
+        hybrid.InterruptableTabuSampler(),
+        subproblem | subsampler
+    ) | hybrid.ArgMin()
+
+    workflow = hybrid.LoopUntilNoImprovement(iteration, max_iter=1000, convergence=3)
```

Comment:
We can now even omit `max_iter`, because it's infinite by default since recently (so the only termination criterion is the convergence; which yes, could be ill defined - imagine cycles of length 4+...)